world/templates/migrations/0002_auto_20181031_1455.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.16 on 2018-10-31 14:55
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('templates', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AlterModelManagers(
+            name='template',
+            managers=[
+            ],
+        ),
+    ]
```

Comment:
So the only other thing I'm thinking about is it might be a good idea to do a data migration to convert existing books into the template system. If you haven't written data migrations in django before, you'd define a function that has the arguments of `(apps, schema_editor)` for `migrations.RunPython` to take as an argument. Inside the function, you'd need to fetch a workable version of `Attribute` with `Attribute = apps.get_model("typeclasses", "Attribute")`, then get descs for Readables with like `qs = Attribute.objects.filter(db_key="desc", objectdb__typeclass_path="typeclasses.readable.readable.Readable")`.

Once you have those descs, you can start to build templates from them by looking at each Attribute's db_value, making a template from it if it's new, and then setting the Attribute's db_value to be `"[[TEMPLATE_%s]]" % new_template.id`.

If that sounds like too much to mess with, I've done that stuff a lot before so I wouldn't mind whipping it up, but if you wanna give it a whirl, by all means.

Reply:
Probably has to be the Attribute db_strvalue field, though. Otherwise the template's markup characters would probably make it into a pickled list or whatever.

Reply:
I think we need to be careful about this, especially if we are going to support 'owners'.

We need to reconcile around who owned the first instance, if they are banned players, etc.

Structurally, I had thought of a fourth 'ACCESS' level, which is like 'UNASSIGNED', and is attached to a root PlayerAccount that has been built for the purposes of serving as an owner for that, for auditing.

However, I think seeing the system in flight first is a wiser idea.

---

I was thinking of building an admin tool that would do an audit of a typeclasses (arbitrarily's) desc characteristics and return a menu of name and desc, along with details regarding the individuals that have made copies of these duplicates.

Then, being able to, by command go 'reconcile'. The advantage of this approach is being able to continually audit as necessary. It does place a certain additional onus on staff, but, given the fact that it assigns access to player accounts, I think there is some value in waiting.

Taking some time to test the system first doesn't preclude the ability to run the appropriate migrations.

Reply:
I was looking at this and it'd be more tricky to do this as a migration than I thought, because the serialization/deserialization steps that Evennia does probably wouldn't work in the migration itself, forcing us to do that manually - the descs are stored in `db_value`, not `db_strvalue`, which seems like a bug. 

So probably building an admin tool might be good approach, yeah. If you're satisfied with how things look now, I think I can go ahead and merge this and we can see how things shake out before thinking of how to convert existing books to using templates.

Reply:
I'm down. :)